openshift/system/config/redis.yml

```
@@ -5,3 +5,12 @@ production:
   sentinels: "<%= ENV['REDIS_SENTINEL_HOSTS'] %>"
   name: <%= ENV['REDIS_SENTINEL_NAME'] %>
   role: <%= ENV['REDIS_SENTINEL_ROLE'] %>
+  # == ACL Params ==
+  username: <%= ENV['REDIS_ACL_USERNAME'] %>
+  password: <%= ENV['REDIS_ACL_PASSWORD'] %>
+  # == TLS Params ==
+  ssl_params:
+    ca_file: <%= ENV['REDIS_TLS_CA_FILE'] %>
+    cert: <%= ENV['REDIS_TLS_CERT'] %>
+    key: <%= ENV['REDIS_TLS_KEY'] %>
```

```suggestion
    cert: <%= ENV['REDIS_USER_CERT'] %>
    key: <%= ENV['REDIS_PRIVATE_KEY'] %>
```

This naming makes it more clear about the intended purpose of these variables. `TLS` is redundant in this case I believe. What else can certificate be about? I'm not super much against the `TLS` part but it's nicer without I believe.

---

I like the original names (i.e. `REDIS_TLS_CERT` and `REDIS_TLS_KEY`), because they match how Redis itself calls them, see https://redis.io/docs/management/security/encryption/#running-manually

Also, isn't it called "client certificate" rather than "user"?... But maybe I'm misinterpreting what `ssl_params.cert` represents :grimacing:

---

> they match how Redis itself calls them, see https://redis.io/docs/management/security/encryption/#running-manually

They match the server parameters names, but we are the client here.

> Also, isn't it called "client certificate" rather than "user"?... But maybe I'm misinterpreting what `ssl_params.cert` represents 😬

As Porta is the client, `cert` is the client certificate.

What about these options?
- `REDIS_CA_FILE`, `REDIS_CERT` and `REDIS_KEY`
  - To match params names
- `REDIS_CLIENT_CA_FILE`, `REDIS_CLIENT_CERT` and `REDIS_CLIENT_KEY`
  - To note it's the client, and match params names

---

> What about these options?
> 
> * `REDIS_CA_FILE`, `REDIS_CERT` and `REDIS_KEY`
>   
>   * To match params names
> * `REDIS_CLIENT_CA_FILE`, `REDIS_CLIENT_CERT` and `REDIS_CLIENT_KEY`
>   
>   * To note it's the client, and match params names

I think `CLIENT_CA_FILE` is not necessarily accurate, it's not client or server, it's just CA, isn't it? :upside_down_face: 

`REDIS_CA_FILE` and `REDIS_CERT` look good, but `REDIS_KEY` IMO is too vague.

So, I'd be sticking with the original options. Optionally, `_TLS_` could be replaced with `_SSL_`, but probably not good, because TLS term kind of supersedes SSL.

---

> I think `CLIENT_CA_FILE` is not necessarily accurate, it's not client or server, it's just CA, isn't it? 🙃

Right!! 😅

> `REDIS_CA_FILE` and `REDIS_CERT` look good, but `REDIS_KEY` IMO is too vague.
> 
> So, I'd be sticking with the original options.

What do you mean? Do you have an example of variable names?

---

> What do you mean? Do you have an example of variable names?

I mean that I wouldn't change what you have now :)

```
  ssl_params:
   ca_file: <%= ENV['REDIS_TLS_CA_FILE'] %>
   ca_path: <%= ENV['REDIS_TLS_CA_PATH'] %>
   cert: <%= ENV['REDIS_TLS_CERT'] %>
   key: <%= ENV['REDIS_TLS_KEY'] %>
```

---

True, "client certificate". For me `TLS` is redundant as I said but I can live with it. More important for me is to have `CLIENT_CERT` and `PRIVATE_KEY`. This makes things clear beyond doubt.

We don't have to follow redis naming conventions rather we better establish a clear configuration naming conventions for porta.

---

I finally set them to

```yaml
 ssl_params:
   ca_file: <%= ENV['REDIS_CA_FILE'] %>
   ca_path: <%= ENV['REDIS_CA_PATH'] %>
   cert: <%= ENV['REDIS_CLIENT_CERT'] %>
   key: <%= ENV['REDIS_PRIVATE_KEY'] %>
```
I hope you both agree on these.

---

I'm fine with these.